pynetsnmp/netsnmp.py

```diff
@@ -472,7 +472,7 @@ def awaitTraps(self, peername, fileno = -1, pre_parse_callback=None, debug=False
             lib.netsnmp_udp6_ctor()
         else:
             log.debug("Cannot find constructor function for UDP/IPv6 transport domain object.")
-        lib.init_snmpv3(None)
+        lib.init_snmp("zenoss_app")
```

Comment:
`init_snmp` from `snmp_api.c` contains `init_snmpv3(type);` as well.
In this way of initialization `msgAuthoritativeEngineBoots` and `msgAuthoritativeEngineTime` are set up properly so traps in v3 can be authenticated correctly.